components/webvr/webvr_thread.rs

```diff
@@ -230,16 +250,20 @@ impl WebVRThread {
         sender.send(self.polling_events).unwrap();
     }
 
-    fn notify_events(&self, events: Vec<VRDisplayEvent>) {
+    fn notify_events(&self, mut events: Vec<VREvent>) {
         let pipeline_ids: Vec<PipelineId> = self.contexts.iter().map(|c| *c).collect();
-        for event in events {
-            let event = WebVREventMsg::DisplayEvent(event);
-            self.constellation_chan.send(ConstellationMsg::WebVREvent(pipeline_ids.clone(), event)).unwrap();
-        }
+        let events = events.drain(0..).map(|ev| {
+            match ev {
+                VREvent::Display(ev) => WebVREventMsg::DisplayEvent(ev),
+                VREvent::Gamepad(ev) => WebVREventMsg::GamepadEvent(ev),
+            }
```

Comment:
I did that to have the possibility to add some internal events in the future. But it's not needed now, so I'll change that only if required in the future.